util/circlog/circlog_test.go

```diff
@@ -0,0 +1,163 @@
+package circlog_test
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/zrepl/zrepl/util/circlog"
+)
+
+func TestCircularLog(t *testing.T) {
+	var maxCircularLogSize int = 1 << 20
+
+	writeFixedSize := func(w io.Writer, size int) (int, error) {
+		pattern := []byte{0xDE, 0xAD, 0xBE, 0xEF}
+		writeBytes := bytes.Repeat(pattern, size/4)
+		if len(writeBytes) < size {
+			writeBytes = append(writeBytes, pattern[:size-len(writeBytes)]...)
+		}
+		n, err := w.Write(writeBytes)
+		if err != nil {
+			return n, err
+		}
+		/*
```

Comment:
why is this in here?

Reply:
Because the `writeFixedSize` closure wraps the `io.Writer` interface that is passed to it, so it returns the same values that `Writer.Write()` would. I could have put `require.NoError()` inside the closure, but I prefer to keep all the assertions farther down in the test body, where I call `writeFixedSize()`.

Reply:
I was referring to the commented-out code block...

Reply:
I must have been tired when reviewing your comments. Of course, that's some dead code that needs to be taken out. Fixed.